packages/angular/projects/clr-angular/src/forms/checkbox/checkbox-container.ts

```diff
@@ -31,18 +40,20 @@ import { LayoutService } from '../common/providers/layout.service';
     '[class.clr-form-control-disabled]': 'control?.disabled',
     '[class.clr-row]': 'addGrid()',
   },
-  providers: [NgControlService, ControlClassService, IfErrorService],
+  providers: [NgControlService, IfSuccessService, ControlClassService, IfErrorService],
 })
 export class ClrCheckboxContainer extends ClrAbstractContainer {
   private inline = false;
+  @ContentChild(ClrControlSuccess) controllSuccessComponent: ClrControlSuccess;
```

Comment:
Initially, we didn't need this for the error state - but because the way we get is something success or not, is by asking the NgControl and will we get `valid` on touched elements that don't have any validation rules. 

This is triggering the success icon to be shown. The way that I come up with knowing when the user wants to have the success icon and message was if the `clr-control-success` is present. If not the green success icon won't be visible when the form element is valid.